packages/fx-core/src/plugins/solution/fx-solution/solution.ts

```diff
@@ -641,10 +643,11 @@ export class TeamsAppSolution implements Solution {
 
       // Only Azure project requires this confirm dialog
       const username = (azureToken as any).username ? (azureToken as any).username : "";
-      const subscriptionInfo = await ctx.azureAccountProvider?.getSelectedSubscription();
+      const subscriptionId = ctx.envInfo.profile.get(GLOBAL_CONFIG)?.get(SUBSCRIPTION_ID) as string;
```

Comment:
If user didn't config the sub id in config.env.json, can we use the one from account provider?

Reply:
Yes. This PR is only a refactor, it does not change business logic. In commonQuestions.ts, solution has already put `subscriptionId` into `ctx`, and the plugins only need to read from it.